fs/expose/dokan/__init__.py

```diff
@@ -731,36 +783,66 @@ def LockFile(self, path, offset, length, info):
             except KeyError:
                 locks = self._active_locks[path] = []
             else:
-                errno = self._check_lock(path, offset, length, None, locks)
-                if errno:
-                    return errno
+                status = self._check_lock(path, offset, length, None, locks)
+                if status:
+                    return status
             locks.append((info.contents.Context, offset, end))
-            return 0
+            return STATUS_SUCCESS
 
     @timeout_protect
     @handle_fs_errors
     def UnlockFile(self, path, offset, length, info):
-        end = offset + length
         with self._files_lock:
             try:
                 locks = self._active_locks[path]
             except KeyError:
-                return -ERROR_NOT_LOCKED
+                return STATUS_NOT_LOCKED
             todel = []
             for i, (lh, lstart, lend) in enumerate(locks):
                 if info.contents.Context == lh:
                     if lstart == offset:
                         if lend == offset + length:
                             todel.append(i)
             if not todel:
-                return -ERROR_NOT_LOCKED
+                return STATUS_NOT_LOCKED
             for i in reversed(todel):
                 del locks[i]
-            return 0
+            return STATUS_SUCCESS
+
+    @handle_fs_errors
+    def GetFileSecurity(self, path, securityinformation, securitydescriptor, securitydescriptorlength, neededlength, info):
+        securitydescriptor = ctypes.cast(securitydescriptor, libdokan.PSECURITY_DESCRIPTOR)
+        path = self._dokanpath2pyfs(path)
+        if self.fs.isdir(path):
+            res = libdokan.GetFileSecurity(
+                self.securityfolder,
+                ctypes.cast(securityinformation, libdokan.PSECURITY_INFORMATION)[0],
+                securitydescriptor,
+                securitydescriptorlength,
+                neededlength,
+            )
+            return STATUS_SUCCESS if res else STATUS_BUFFER_OVERFLOW
+        return STATUS_NOT_IMPLEMENTED
+
+    @handle_fs_errors
+    def SetFileSecurity(self, path, securityinformation, securitydescriptor, securitydescriptorlength, info):
+        return STATUS_NOT_IMPLEMENTED
+
+    @handle_fs_errors
+    def Mounted(self, info):
+        return STATUS_SUCCESS
+
+    @handle_fs_errors
+    def Unmounted(self, info):
+        return STATUS_SUCCESS
```

Comment:
@lurch This functions are part of dokan API and are descibred there.
Otherwise there is some callback https://github.com/PyFilesystem/pyfilesystem/blob/fc74df6ababbd48d79c09c58e326f2b40293dbb7/fs/expose/dokan/__init__.py#L875 that are already documented